### PR TITLE
Fixed microdnf support

### PR DESCRIPTION
**Pull request by schaefi**
The installroot argument must be used together with --config
and additionally with --noplugins, as well as --setopt for
cachedir, reposdir and varsdir. Related to #1625

### CODE REVIEW

File: kiwi/repository/dnf.py [hunk 1]
```diff
@@ -80,7 +80,8 @@ def post_init(self, custom_args=None):
         self.shared_dnf_dir = {
             'reposd-dir': manager_base + '/repos',
             'cache-dir': manager_base + '/cache',
-            'pluginconf-dir': manager_base + '/pluginconf'
+            'pluginconf-dir': manager_base + '/pluginconf',
+            'vars-dir': manager_base + '/vars'
```
Comment (Conan-Kudo) [inline kiwi/repository/dnf.py:84]: We should add `varsdir` setting to the generated `dnf.conf` file, as that can be used by DNF too, and I could see reasons for us to want to populate that in the future.

**Review by Conan-Kudo — CHANGES_REQUESTED** [review]

File: kiwi/package_manager/microdnf.py [hunk 2]
```diff
@@ -102,7 +102,16 @@ def process_install_requests_bootstrap(self, root_bind=None):
         bash_command = [
             'microdnf'
         ] + ['--refresh'] + self.dnf_args + [
-            '--installroot', self.root_dir
+            '--installroot', self.root_dir, '--noplugins',
+            '--setopt=cachedir={0}'.format(
+                self.repository.shared_dnf_dir['cache-dir']
```
Comment (Conan-Kudo) [inline kiwi/package_manager/microdnf.py:107]: Does `self.repository` exist at this point?

**Review by Conan-Kudo — CHANGES_REQUESTED** [review]

File: kiwi/repository/dnf.py (hunk 1 above)
Comment (schaefi) [reply kiwi/repository/dnf.py:84]: ok will adapt regarding this

File: kiwi/package_manager/microdnf.py (hunk 2 above)
Comment (schaefi) [reply kiwi/package_manager/microdnf.py:107]: sure, the package manager object can only be created by passing in a repository instance

Comment (Conan-Kudo) [reply kiwi/package_manager/microdnf.py:107]: Ah, okay.

Comment (schaefi) [reply kiwi/package_manager/microdnf.py:107]:
```python
class PackageManagerMicroDnf(PackageManagerBase):
```

and the base class has

```python
def __init__(self, repository, custom_args=None):
```

File: kiwi/repository/dnf.py (hunk 1 above)
Comment (schaefi) [reply kiwi/repository/dnf.py:84]: done

**Review by Conan-Kudo — APPROVED** [review]

**Pull request merged**